src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -5172,7 +5172,7 @@ operand indOffX2P(iRegL reg, immLOffset off)
 
 operand indirectN(iRegN reg)
 %{
-  predicate(CompressedOops::shift() == 0);
+  predicate(CompressedOops::shift() == 0 && !SCCache::is_on_for_write());
```

Comment:
Does it imply that with SCC is on_write we will not have the shift == 0?

Reply:
No, this is to prevent the match when compiling AOT code. We don't want to generate any AOT code which depends on shift == 0 as it may change during runtime.

Reply:
Okay.